### ISSUE TITLE
Mutation updates, appending to lists, etc

### ISSUE THREAD

**Issue by rbalicki2** [isographlabs/isograph#278]

**Comment by Benanna2019** [comment]
I know I am really new to this but could we use the useOptimistic hook from react and maybe wrap that with something Isograph specific so that it updates the isograph store optimistically and then has a fallback if the mutation errors?

**Comment by rbalicki2** [comment]
Great question. I think yes, you can do that. As in, nothing will prevent `useOptimistic` from working, AFAIK.

However, the isograph store should be the source of truth for optimistic state, as this ensures consistency throughout the app (e.g. if you optimistically update your name, it will also update your name in the header).

In addition, Isograph can be more sophisticated that what you could do with `useOptimistic`. Consider, for example, a scenario in which you optimistically updated whether you like a stock. While that request is in flight, the stock price changed (e.g. we have a subscription to a price ticker). Isograph can (and will) re-apply both updates, rendering `{ isLiked: newIsLiked, price: newPrice }`. OTOH if you used `useOptimistic`, you would materialize at some point `{ ...stonk, isLiked: newIsLiked,  }`, which would stop reflecting changes in the price.

Basically, Relay has a great model for this and we're going to re-implement that :)

**Comment by rbalicki2** [comment]
Oh and optimistic updates need to work with non-React frameworks, too!
